src/NinjaTools.FluentMockServer/MockServerClient.cs

```diff
@@ -47,41 +50,54 @@ public Task SetupAsync(Func<IFluentExpectationBuilder, MockServerSetup> setupFac
         /// <returns></returns>
         public async Task SetupAsync(MockServerSetup setup)
         {
-            foreach (var request in setup.Expectations.Select(Expectation))
+            foreach (var request in setup.Expectations.Select(GetExpectationMessage))
             {
-                var response = await _httpClient.SendAsync(request);
+                var response = await HttpClient.SendAsync(request);
                 response.EnsureSuccessStatusCode();
             }
         }
 
         public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request)
         {
-            var response = await _httpClient.SendAsync(request);
+            var response = await HttpClient.SendAsync(request);
             return response;
         }
 
         public async Task<HttpResponseMessage> ResetAsync()
         {
-            var request = Reset();
-            var response = await _httpClient.SendAsync(request);
+            var request = GetResetMessage();
+            var response = await HttpClient.SendAsync(request);
             response.EnsureSuccessStatusCode();
             return response;
         }
 
         public async Task<HttpResponseMessage> VerifyAsync(Verify verify)
         {
-            var request = Verify(verify);
-            var response = await _httpClient.SendAsync(request);
-            response.EnsureSuccessStatusCode();
+            var request = GetVerifyRequest(verify);
+            var response = await HttpClient.SendAsync(request);
             return response;
         }
-        
+
+        public async Task<(bool isValid, string responseMessage)> VerifyAsync(Action<IFluentVerificationBuilder> verify)
+        {
+            var response = await Verify(verify);
+            var responseMessage = await response.Content.ReadAsStringAsync();
+            return (response.StatusCode == HttpStatusCode.Accepted, responseMessage);
+        }
+
+        public Task<HttpResponseMessage> Verify(Action<IFluentVerificationBuilder> verify)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename the parameter 'verify' so that it does not duplicate the method name.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4812987)